lib/submit-analytics.js

```diff
@@ -33,50 +120,163 @@ function sha256(data) {
     return hash.digest('hex');
 }
 
-if (isAnalyticsDisabled()) {
-    module.exports = function(){};
-} else {
+/**
+ * Send collected analytics data to Realm's servers over HTTPS
+ * @param  {[WebhookProdDetails|WebhookStageDetails|MixpanelDetails]} requestFactory Submission URL generator
+ * @param  {Object} payload analytics info
+ */
+async function dispatchAnalytics(requestFactory, payload) {
+    const https = require('https');
+
+    return new Promise((resolve, reject) => {
+        const webhookRequest = new requestFactory().buildRequest(payload);
+        var req = https.get(webhookRequest, function (result) {
+            result.on('data', (data) => {
+                // we don't log the data that is returned -- we're just interested in the header
+            });
+        });
```

Comment:
What's the reason for adding the listener and callback them?

```suggestion
        var req = https.get(webhookRequest);
```